content/en/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver.md

```diff
@@ -0,0 +1,124 @@
+---
+title: Configuring a cgroup driver
+content_type: task
+weight: 10
+---
+
+<!-- overview -->
+
+This page explains how to configure the kubelet cgroup driver to match the container
+runtime cgroup driver for kubeadm clusters.
+
+## {{% heading "prerequisites" %}}
+
+You should be familiar with the Kubernetes
+[container runtime requirements](/docs/setup/production-environment/container-runtimes).
+
+<!-- steps -->
+
+## Configuring the container runtime cgroup driver
+
+The [Container runtimes](/docs/setup/production-environment/container-runtimes) page
+provides details on how to setup a number of different container runtimes with the
+`systemd` driver by default.
+
+The same page explains that the `systemd` driver is recommended for kubeadm based setups
+instead of the `cgroupfs` driver, because kubeadm manages the kubelet as a systemd service.
+
+## Configuring the kubelet cgroup driver
+
+kubeadm allows you to pass a `KubeletConfiguration` structure during `kubeadm init`.
+This `KubeletConfiguration` can include the `cgroupDriver` field which controls the cgroup
+driver of the kubelet.
+
+{{< note >}}
+If the user is not setting the `cgroupDriver` field under `KubeletConfiguration`,
+`kubeadm init` will default it to `systemd`.
+{{< /note >}}
```

Comment:
this is new to kubeadm 1.21

Reply:
Let's add "starting for v1.21" and "for older releases, the user should executed following steps manually"

Reply:
i think the point of the versioned docs (this PR is targeting dev-1.21) is that all details apply to the target release - i.e. 1.21.

maybe we can use some sort of version tag for this.
will defer to sig-docs for this question.

Reply:
added: `{{< feature-state for_k8s_version="v1.21" state="stable" >}}` inside the note.
not sure if that is the preferred way.